### PR TITLE
Platform security 20230919

### PR DESCRIPTION
**Pull request by pyth0n1c**
Detections for Platform Security

### CODE REVIEW

File: detections/application/splunk_absolute_path_traversal_using_runshellscript.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Splunk Absolute Path Traversal Using runshellscript
+id: 356bd3fe-f59b-4f64-baa1-51495411b7ad
+version: 1
+date: '2023-09-05'
+author: Rod Soto
+status: production
+type: Hunting
+data_source:
+- python.log
```
Comment (patel-bhavin) [inline detections/application/splunk_absolute_path_traversal_using_runshellscript.yml]: this should be empty

File: detections/application/splunk_absolute_path_traversal_using_runshellscript.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+known_false_positives: The command runshellscript can be used for benign purposes. Analyst will have to review the searches and determined maliciousness specially by looking at targeted script.
+references:
+- https://advisory.splunk.com/advisories/SVD-2023-0806
+tags:
+  analytic_story:
+  - Splunk Vulnerabilities 
+  asset_type: endpoint
+  confidence: 70
+  impact: 50
+  message: Possible attack against splunk_server $splunk_server$ through abuse of the runshellscript command
```
Comment (patel-bhavin) [inline detections/application/splunk_absolute_path_traversal_using_runshellscript.yml:34]: Lets rename `splunk_server` to `dest` and update this on observables

File: detections/application/splunk_cmd_injection_using_external_lookups.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+id: 901ae2e2-b47b-4adc-b296-a151a8fccdca
+version: 1
+date: '2023-09-05'
+author: Rod Soto
+status: production
+type: Hunting
+data_source:
+- audittrial
+description: In Splunk Enterprise versions below 8.2.12, 9.0.6, and 9.1.1, an attacker can create an external lookup that calls a legacy internal function. The attacker can use this internal function to insert code into the Splunk platform installation directory. From there, a user can execute arbitrary code on the Splunk platform Instance.
+search: '`audit_searches` search = "| *runshellscript* """ | stats count min(_time) as firstTime max(_time) as lastTime by search user splunk_server
```
Comment (patel-bhavin) [inline detections/application/splunk_cmd_injection_using_external_lookups.yml]: Lets rename `host` to `dest` and update this on observables and message and this search is currently failing CI

File: detections/application/splunk_dos_using_malformed_saml_request.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+id: 8e8a86d5-f323-4567-95be-8e817e2baee6
+version: 1
+date: '2023-09-05'
+author: Rod Soto
+status: production
+type: Hunting
+data_source:
+- splunkd.log 
+description: In Splunk Enterprise versions lower than 9.0.6, and 8.2.12, an attacker can send a malformed security assertion markup language SAML request to the /saml/acs REST endpoint which can cause a denial of service through a crash or hang of the Splunk daemon.The SAML extensible markup language (XML) parser does not fail SAML signature validation when the attacker modifies the URI in the SAML request. Instead it attempts to access the modified URI, which causes the Splunk daemon to crash or hang.
+search: '`splunkd` event_message=*error* expr=*xpointer* | stats count min(_time) as firstTime max(_time) as lastTime by component expr host event_message | `security_content_ctime(firstTime)`| `security_content_ctime(lastTime)` | `splunk_dos_using_malformed_saml_request_filter`'
```
Comment (patel-bhavin) [inline detections/application/splunk_dos_using_malformed_saml_request.yml]: Lets rename host to dest and update this on observables and message

File: detections/application/splunk_reflected_xss_on_app_search_table_endpoint.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+id: 182f9080-4137-4629-94ac-cb1083ac981a
+version: 1
+date: '2023-09-05'
+author: Rod Soto
+status: production
+type: Hunting
+data_source:
+- web_access.log 
+description: In Splunk Enterprise versions below 9.1.1, 9.0.6, and 8.2.12, an attacker can craft a special web request that can result in reflected cross-site scripting XSS on the app search table web endpoint, which presents as the Create Table View page in Splunk Web. Exploitation of this vulnerability can lead to the execution of arbitrary commands on the Splunk platform instance. A JavaScript file within this web endpoint does not properly validate input which lets an attacker insert a payload into a function.
+search: '`splunkd_web` (dataset_commands="*makeresults*" AND dataset_commands="*count*" AND dataset_commands="*eval*" AND dataset_commands="*baseSPL*") | stats count min(_time) as firstTime max(_time) as lastTime by clientip status user view root uri_path | `security_content_ctime(firstTime)`| `security_content_ctime(lastTime)` | `splunk_reflected_xss_on_app_search_table_endpoint_filter`'
```
Comment (patel-bhavin) [inline detections/application/splunk_reflected_xss_on_app_search_table_endpoint.yml:10]: i dont see a host related field in the SPL, can we perhaps add that? and then rename host to dest

File: detections/application/splunk_absolute_path_traversal_using_runshellscript.yml (hunk 2 above)
Comment (patel-bhavin) [reply detections/application/splunk_absolute_path_traversal_using_runshellscript.yml:34]: ignore

File: detections/application/splunk_cmd_injection_using_external_lookups.yml (hunk 3 above)
Comment (patel-bhavin) [reply detections/application/splunk_cmd_injection_using_external_lookups.yml:11]: ignore

Comment (pyth0n1c) [reply detections/application/splunk_cmd_injection_using_external_lookups.yml]: After followup conversation, we will keep this as splunk_server for these detections.

**Review by rosplk — APPROVED** [review]

**Review by patel-bhavin — APPROVED** [review]

**Pull request merged**